Sources/Search/APIs/Custom.php

```diff
@@ -337,6 +364,377 @@ public function postModified(array &$msgOptions, array &$topicOptions, array &$p
 			}
 		}
 	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function postRemoved(int $id_msg): void
```

Comment:
this should be added to `supportsMethod()`

Reply:
Good catch

Reply:
Looking further into this confusing mess, tit appears as if it returns  false, it bubbles up to the parent

https://github.com/SimpleMachines/SMF/blob/b11d8c4178b3d8e18a5243c33174524d216b8e4e/Sources/Search/APIs/Custom.php#L91-L99

This interface violates the Interface Segregation Principle. Perhaps we can break out the post and the topic methods into their own interfaces.

Reply:
> confusing mess

Indeed. The search APIs date way back to the early days of SMF 2.1, and they were less than rigorously structured, to say the least. This PR is working toward cleaning up the mess, but there's still going to be more to do after this.